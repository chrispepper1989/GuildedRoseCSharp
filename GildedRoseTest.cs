using System;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using csharp;

namespace GildedRoseCSharp
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class GildedRoseTest
    {

        //generate a bunch of test cases
        public IList<Item> GenerateTestItems(string[] names, int[] sellIns, int[] qualities)
        {
            IList<Item> Items = new List<Item>
            {
            };
            foreach (var name in names)
            {
                foreach (var sellIn in sellIns)
                {
                    foreach (var quality in qualities)
                    {
                        var item = new Item{ Name =name, Quality = quality, SellIn = sellIn};
                        Items.Add( item);
                    }

                }

            }

            return Items;

        }

         [Test]
         public void Test2()
         {


             //arrange
             var Items = GenerateTestItems( new [] { "Aged Brie" }, new []{1, 5,10,2,4}, new[] { 1, 2 });
             GildedRose app = new GildedRose(Items);
             //act
             app.UpdateQuality();
             //assert
             Approvals.VerifyAll(Items,"All Items");

         }
    }
}
