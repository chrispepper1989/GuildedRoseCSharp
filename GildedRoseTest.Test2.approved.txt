All Items[0] = Aged Brie, 0, 2
All Items[1] = Aged Brie, 0, 3
All Items[2] = Aged Brie, 4, 2
All Items[3] = Aged Brie, 4, 3
All Items[4] = Aged Brie, 9, 2
All Items[5] = Aged Brie, 9, 3
All Items[6] = Aged Brie, 1, 2
All Items[7] = Aged Brie, 1, 3
All Items[8] = Aged Brie, 3, 2
All Items[9] = Aged Brie, 3, 3
